Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.cpp

```diff
@@ -1478,24 +1481,23 @@ void TerrainSystem::OnTick(float /*deltaTime*/, AZ::ScriptTimePoint /*time*/)
 
         // This needs to happen before the "system active" check below, because activating the system will cause the various
         // terrain layer areas to request the current world bounds.
-        if (m_requestedSettings.m_worldBounds != m_currentSettings.m_worldBounds)
+        if (m_requestedSettings.m_heightRange != m_requestedSettings.m_heightRange)
         {
-            m_dirtyRegion = m_currentSettings.m_worldBounds;
-            m_dirtyRegion.AddAabb(m_requestedSettings.m_worldBounds);
+            m_dirtyRegion = m_cachedAreaBounds;
             m_terrainHeightDirty = true;
             m_terrainSurfacesDirty = true;
-            m_currentSettings.m_worldBounds = m_requestedSettings.m_worldBounds;
+            m_currentSettings.m_heightRange = m_requestedSettings.m_heightRange;
         }
 
         if (m_requestedSettings.m_heightQueryResolution != m_currentSettings.m_heightQueryResolution)
         {
-            m_dirtyRegion.AddAabb(m_requestedSettings.m_worldBounds);
+            m_dirtyRegion = m_cachedAreaBounds;
             m_terrainHeightDirty = true;
         }
 
         if (m_requestedSettings.m_surfaceDataQueryResolution != m_currentSettings.m_surfaceDataQueryResolution)
         {
-            m_dirtyRegion.AddAabb(m_requestedSettings.m_worldBounds);
+            m_dirtyRegion = m_cachedAreaBounds;
             m_terrainSurfacesDirty = true;
         }
```

Comment:
These should probably also use the same clamping logic as I describe above if it's going to replace m_dirtyRegion, since these are non-exclusive if statements and the height range may have also changed.

Alternatively, they could use AddAabb(GetTerrainAabb()) to add the current Aabb clamped to the current height range.